src/lib/create.c

```diff
@@ -1055,16 +1053,27 @@ pgp_write_litdata(pgp_output_t *         output,
                   const int              maxlen,
                   const pgp_litdata_enum type)
 {
-    /*
-     * RFC4880 does not specify a meaning for filename or date.
-     * It is implementation-dependent.
-     * We will not implement them.
-     */
-    /* \todo do we need to check text data for <cr><lf> line endings ? */
+    char *   filename = NULL;
+    uint64_t modtime = 0;
+    unsigned flen = 0;
+    /* \todo do we need to check text data for <cr><lf> line endings ? - Yes, we need.
+    For non-PGP_LDT_BINARY we should convert line endings to the canonical CRLF style. */
+
+    if (output->ctx) {
+        filename = output->ctx->filename;
+        modtime = output->ctx->filemtime;
+        flen = filename ? strlen(filename) : 0;
+        if (flen > 255) {
+            (void) fprintf(stderr, "pgp_write_litdata : filename %s too long\n", filename);
+            return 0;
+        }
+    }
+
     return pgp_write_ptag(output, PGP_PTAG_CT_LITDATA) &&
-           pgp_write_length(output, (unsigned) (1 + 1 + 4 + maxlen)) &&
-           pgp_write_scalar(output, (unsigned) type, 1) && pgp_write_scalar(output, 0, 1) &&
-           pgp_write_scalar(output, 0, 4) && pgp_write(output, data, (unsigned) maxlen);
+           pgp_write_length(output, (unsigned) (1 + 1 + flen + 4 + maxlen)) &&
```

Comment:
Here it is written in this way to easier check whether it conforms to the RFC packet structure, seems pretty clear to me.